src/find_library.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rcpputils/find_library.hpp"
+
+#include <cassert>
+#include <cstddef>
+
+#include <list>
+#include <fstream>
+#include <sstream>
+#include <string>
+
+namespace rcpputils
+{
+
+namespace
+{
+
+#ifdef _WIN32
+static constexpr char kPathVar[] = "PATH";
+static constexpr char kPathSeparator = ';';
+static constexpr char kSolibPrefix[] = "";
+static constexpr char kSolibExtension[] = ".dll";
+#elif __APPLE__
+static constexpr char kPathVar[] = "DYLD_LIBRARY_PATH";
+static constexpr char kPathSeparator = ':';
+static constexpr char kSolibPrefix[] = "lib";
+static constexpr char kSolibExtension[] = ".dylib";
+#else
+static constexpr char kPathVar[] = "LD_LIBRARY_PATH";
+static constexpr char kPathSeparator = ':';
+static constexpr char kSolibPrefix[] = "lib";
+static constexpr char kSolibExtension[] = ".so";
+#endif
+
+std::string get_env_var(const char * kPathVar)
+{
+  char * value = getenv(kPathVar);
+  return value ? value : "";
+}
+
+std::list<std::string> split(const std::string & value, const char delimiter)
+{
+  std::list<std::string> list;
+  std::istringstream ss(value);
+  std::string s;
+  while (std::getline(ss, s, delimiter)) {
+    list.push_back(s);
+  }
+  return list;
+}
+
+bool is_file_exist(const char * filename)
+{
+  std::ifstream h(filename);
+  return h.good();
+}
+
+}  // namespace
+
+std::string find_library_path(const std::string & library_name)
+{
+  // TODO(eric.cousineau): Does Poco provide this functionality
+  // (ros2/rcpputils#7)?
+  // TODO(eric.cousineau): Have this return a library pointer instead
+  // (ros2/rcpputils#8).
```

Comment:
What about these TODOs?

Reply:
Sorry, it's unclear what the defect is; can I ask what you'd like to know or how you'd like this to change?

Reply:
(Do you want them removes since they are captured as issues?)

Reply:
Yes please, since the tickets are enhancements tracking them as ticket alone should be enough.

Reply:
Done.